src/UCTSearch.cpp

```diff
@@ -137,6 +137,54 @@ void UCTSearch::dump_stats(KoState & state, UCTNode & parent) {
     }
 }
 
+bool UCTSearch::should_resign(passflag_t passflag) {
+    if (passflag & UCTSearch::NORESIGN) {
+        // resign not allowed
+        return false;
+    }
+
+    const auto visits = m_root.get_visits();
+    if (visits < 500)  {
+        // low visits
+        return false;
+    }
+
+    const size_t board_squares = m_rootstate.board.get_boardsize()
+                               * m_rootstate.board.get_boardsize();
+    const auto move_threshold = board_squares / 4;
+    const auto movenum = m_rootstate.get_movenum();
+    if (movenum <= move_threshold) {
+        // to early in game to resign
+        return false;
+    }
+
+    const auto color = m_rootstate.board.get_to_move();
+    const auto best_eval = m_root.get_first_child()->get_eval(color);
+    auto resign_threshold = 0.01 * cfg_resignpct;
+    if (best_eval > resign_threshold) {
+        // eval > cfg_resign
+        return false;
+    }
+
+    const auto is_default_cfg_resign = (cfg_resignpct == 10);
```

Comment:
With the above this would then just be a check for < 0, but be moved above the previous block, which would then use 10 as the default.